automation/src/test/java/DemoQA/Pages/RadioButtonPage.java

```diff
@@ -0,0 +1,69 @@
+package DemoQA.Pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+public class RadioButtonPage extends PageObjectBase {
+	
+	private final String url = "https://demoqa.com/radio-button";
+	private RadioButtonControlExtension controller = new RadioButtonControlExtension(this.driver);
+	
+	@FindBy(how=How.ID, using="yesRadio")
+	private WebElement yesRadioElement;
+	
+	@FindBy(how=How.ID, using="impressiveRadio")
+	private WebElement impressiveRadioElement;
+	
+	@FindBy(how=How.ID, using="noRadio")
+	private WebElement noRadioElement;
+	
+	public RadioButtonPage(WebDriver driver) {
+		super(driver);
+	}
+
+	public String getURL() {
+		return url;
+	}
+	
+	public ControlExtensionBase getController() {
+		return this.controller;
+	}
+	
+	public PageObjectBase navigate() {
+		this.driver.navigate().to(url);
+		return this;
+	}
+	
+	public void select(String label) {
+		if (label == "yesRadio") {

```

Comment:
This pattern would violate OpenClosed Principle. As more radio buttons are added you would need to update this list. You could find the Radio button element based on the label in the UI.